### PR TITLE
371 Added UniqueConstraint for external pkeys

### PR DESCRIPTION
**Pull request by adp-atea**
This is not finished, still need to add Error message when triggering IntegrityError. IntegrityError handle has been started in task 363: https://github.com/atviriduomenys/spinta/issues/363

There I have created try except around insert, update and delete. It checks for IntegrityError, if it finds it, it adds it to the MultipleErrors exception. Also if it finds any errors it reverts the transaction. The upside of that implementation, is that it checks all of the data or until it find certain amount of errors, so user gets more informative response.

With current UniqueConstraint error implementation user only gets 1 response.

### CODE REVIEW

File: spinta/manifests/tabular/helpers.py [hunk 1]
```diff
@@ -1550,6 +1550,9 @@ def _property_to_tabular(
     if prop.access < access:
         return
 
+    remember_unique_state = prop.dtype.unique
+    if len(prop.model.external.pkeys) == 1:
+        prop.dtype.unique = False
```
Comment (sirex) [inline spinta/manifests/tabular/helpers.py:1555]: Please move this into `dtype.get_type_repr`, because this method is responsible for producing correct  type string. Also it would be nice, to move `get_type_repr` out of `dtype`, because, this representation only needed for tabular manifests, so it should be part of tabular manifest, not a high level method.

File: tests/backends/postgresql/commands/test_init.py [hunk 2]
```diff
@@ -32,3 +32,30 @@ def test_prepare(rc: RawConfig):
...
+    assert any(
+        [table.c['id']] == list(constraint.columns) for constraint in table.constraints if
+        type(constraint).__name__ == 'UniqueConstraint')
+
+    model_multiple_unique = manifest.models['example/Country']
+    commands.prepare(context, backend, model_multiple_unique)
+    table = backend.get_table(model_multiple_unique)
+    assert any(
+        [table.c['id'], table.c['name']] == list(constraint.columns) for constraint in table.constraints if
+        type(constraint).__name__ == 'UniqueConstraint')
```
Comment (sirex) [inline tests/backends/postgresql/commands/test_init.py:61]: Please, add tests for handling unique constraint errors, when trying to write a duplicate value on a column or columns, that have unique constraints.

**Review by sirex — CHANGES_REQUESTED** [review]

File: notes/backends/postgresql/pkeys/unique.sh [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+#|                     '_revision': '8de45de1-577d-4ef7-ba50-7480b5d7934f',
+#|                     'id': 1,
+#|                     'code': '01',
+#|                     'title': 'T01'
+#|                 }
+#|             "
+#|         }
+#|     ]
+#| }
+# FIXME: This should be a 400 error.
```
[the remark this answers is not in the document]
Comment (sirex) [reply notes/backends/postgresql/pkeys/unique.sh:180]: Same error here: https://github.com/atviriduomenys/spinta/pull/343#discussion_r1176076447

File: spinta/manifests/tabular/helpers.py (hunk 1 above)
Comment (sirex) [reply spinta/manifests/tabular/helpers.py:1555]: Same issue here: https://github.com/atviriduomenys/spinta/pull/343#discussion_r1176129421

**Comment by sirex** [comment]
Please, do not move pull request to review, if there are unresolved comments.

**Review by sirex — APPROVED** [review]
